cupy/random/_generator_api.pyx

```diff
@@ -351,6 +360,102 @@ class Generator:
 
         return y
 
+    def gamma(self, shape, scale=1.0, size=None):
+        """Returns an array of samples drawn from a gamma distribution.
+
+        Args:
+            shape (float or array_like of float): The shape of the
+                gamma distribution.  Must be non-negative.
+            scale (float or array_like of float): The scale of the
+                gamma distribution.  Must be non-negative.
+                Default equals to 1
+            size (int or tuple of ints): The shape of the array.
+                If ``None``, a zero-dimensional array is generated.
+
+        .. seealso::
+            - :func:`cupy.random.gamma` for full documentation
+            - :meth:`numpy.random.Generator.gamma`
+        """
+        if size is None:
+            size = cupy.broadcast(shape, scale).shape
+        y = self.standard_gamma(shape, size)
+        y *= scale
+        return y
+
+    def standard_gamma(self, shape, size=None, dtype=numpy.float64, out=None):
+        """Returns an array of samples drawn from a standard gamma distribution.
+
+        Args:
+            shape (float or array_like of float): The shape of the
+                gamma distribution.  Must be non-negative.
+            size (int or tuple of ints): The shape of the array.
+                If ``None``, a zero-dimensional array is generated.
+            dtype: Data type specifier.
+            out (cupy.ndarray, optional): If specified, values will be written
+                to this array
+
+        .. seealso::
+            - :func:`cupy.random.standard_gamma` for full documentation
+            - :meth:`numpy.random.Generator.standard_gamma`
+        """
+        cdef ndarray y
+        cdef ndarray shape_arr
+
+        if not isinstance(shape, ndarray):
+            if type(shape) in (float, int):
+                shape_a = ndarray(1, numpy.float64)
+                shape_a.fill(shape)
+                shape = shape_a
+            else:
+                raise ValueError('shape is required to be a cupy.ndarray'
+                                 ' or a scalar')
+        else:
+            # Check if size is broadcastable to shape
+            # but size determines the output
+            shape = shape.astype('d', copy=False)
+
+        if size is not None and not isinstance(size, tuple):
+            size = (size,)
+        elif size is None:
+            size = shape.shape if out is None else out.shape
+
+        internal._broadcast_shapes([size, shape.shape])
```

Comment:
`shape.shape` is checked later at `shape = cupy.broadcast_to(shape, y.shape)`

Reply:
I think I did this to raise some errors in the same order as numpy, but I can't remember which. Anyway I think it's better to delete it :)